backend/tests/e2e/test_api_e2e.py

```
@@ -0,0 +1,988 @@
+                    available_chains = real_alert_service.chain_registry.list_available_chains()
+                    print(f"🔍 Available alert types: {available_alert_types}")
+                    print(f"🔍 Available chains: {available_chains}")
+                    
+                    # Replace the main alert service with our real instance
+                    with patch('tarsy.main.alert_service', real_alert_service):
+                        # STEP 1: Submit Alert 
+                        print("\n📝 STEP 1: Submitting alert...")
+                        response = test_client.post("/alerts", json=realistic_namespace_alert)
+                        if response.status_code != 200:
+                            print(f"❌ Alert submission failed: {response.status_code} - {response.text}")
+                            assert False, f"Alert submission failed: {response.status_code}"
```

Ruff (0.12.2)

Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)